lib/ruby_lsp/requests/support/semantic_token_encoder.rb

```diff
@@ -38,11 +38,22 @@ def compute_delta(token)
           delta_column = column
           delta_column -= @current_column if delta_line == 0
 
-          [delta_line, delta_column, token.length, token.type, token.modifier]
+          [delta_line, delta_column, token.length, token.type, encode_modifiers(token.modifier)]
         ensure
           @current_row = row
           @current_column = column
         end
+
+        # Encode an array of modifiers to positions onto a bit flag
+        # For example, [:default_library] will be encoded as
+        # 0b1000000000, as :default_library is the 10th bit according
+        # to the token modifiers index map.
+        def encode_modifiers(modifiers)
+          modifiers.inject(0) do |encoded_modifiers, modifier|
+            encoded_modifiers |= (1 << modifier)
+            encoded_modifiers
```

Comment:
I think the previous line already returns the result, so you may not need this one.

Reply:
Yeah this can just be:
```ruby
          modifiers.inject(0) do |encoded_modifiers, modifier|
            encoded_modifiers | (1 << modifier)
          end
```
There is no point in mutating `encoded_modifiers` since it is just a local variable. The return value from the block is what is significant and becomes the value of the `encoded_modifiers` for the next loop.